frontend/src/components/ViewActivities/activitylist.js

```diff
@@ -33,25 +48,10 @@ export async function getActivityList(tripId) {
           data[DB.ACTIVITIES_END_TIME]['seconds'] * 1000;
 
         tripActivities.push(data);
-      })
-    }).catch(error => {
-      console.log('It seems that an error has occured.');
-      tripActivities = null;
-    }).then( () => resolve(tripActivities) );
-  })
-}
-
-/**
- * The list of activities. 
- * 
- * @param {Object} props This component expects the following props:
- * - `tripId` {string} The trip's ID.  
- */
-class ActivityList extends React.Component {
-  /** @inheritdoc */
-  constructor(props) {
-    super(props);
-    this.state = { days : [] };
+      });
+      this.setState({ days: activityFns.sortByDate(tripActivities) });
+    });
```

Comment:
If you want, you can pass your error callback into the second parameter of onSnapshot [like so](https://firebase.google.com/docs/firestore/query-data/listen#handle_listen_errors).

Reply:
Yeah. Error behavior is specified in render() though.